src/adapters/ws-pool.ts

```diff
@@ -0,0 +1,109 @@
+import { AppComponents, IWSPoolComponent } from '../types'
+
+export async function createWSPoolComponent({
+  metrics,
+  config,
+  redis
+}: Pick<AppComponents, 'metrics' | 'config' | 'redis'>): Promise<IWSPoolComponent> {
+  const maxConcurrentConnections = (await config.getNumber('MAX_CONCURRENT_CONNECTIONS')) || 100
+  const idleTimeout = (await config.getNumber('IDLE_TIMEOUT')) || 300000 // 5 minutes default
+
+  const cleanupInterval = setInterval(async () => {
+    const now = Date.now()
+    const pattern = 'ws:conn:*'
+
+    for await (const key of redis.client.scanIterator({ MATCH: pattern })) {
+      const data = await redis.client.get(key)
+      if (data) {
+        const parsed = JSON.parse(data)
+        if (now - parsed.lastActivity > idleTimeout) {
```

Comment:
This is a safe check, right? because the TTL set at redis should already be enough to prevent this but maybe it is worth leaving it here

Reply:
Yep, it is just a proactive way of cleaning up idle connections. Redis will eventually clean this up even without this mechanism.